app/views/ActionSheet.js

```diff
@@ -0,0 +1,198 @@
+	snapPoint: SNAP_POINT.HALF,
+	options: {}
+}
+
+option object that you pass should have
+1. label
+2. handler
+3. icon // if not provided gives an flag icon
+
+example optionObject : {
+	label: 'Permalink',
+	handler: this.dummy,
+	icon: 'permalink',
+	isDanger: true
+}
+
+*/
+
+export const LISTENER = 'actionSheet';
+export const SNAP_POINTS = {
+	HIDE: 0,
+	SHORT: 1,
+	HALF: 2,
+	FULL: 3
+};
+
+const bottomSheetHeaderHeight = 25;
+const buttonPaddingSize = verticalScale(35);
+const textSize = 13;
+
+const styles = StyleSheet.create({
+	panel: {
+		paddingVertical: verticalScale(15),
+		paddingHorizontal: verticalScale(10),
+		backgroundColor: COLOR_BACKGROUND_CONTAINER
+	},
+	panelButton: {
+		padding: verticalScale(15),
+		backgroundColor: 'transparent',
+		borderBottomWidth: StyleSheet.hairlineWidth,
+		borderBottomColor: COLOR_SEPARATOR,
+		flexDirection: 'row'
+	},
+	panelButtonIcon: {
+		paddingRight: 10
+	},
+	header: {
+		backgroundColor: COLOR_BACKGROUND_CONTAINER,
+		paddingTop: 15,
+		borderTopLeftRadius: 20,
+		borderTopRightRadius: 20,
+		alignItems: 'center'
+	},
+	panelHandle: {
+		width: 40,
+		height: 8,
+		borderRadius: 4,
+		backgroundColor: COLOR_SEPARATOR
+	},
+	androidButtonView: {
+		borderBottomWidth: StyleSheet.hairlineWidth,
+		borderBottomColor: COLOR_SEPARATOR
+	},
+	buttonText: {
+		...sharedStyles.textRegular,
+		fontSize: 16
+	},
+	danger: {
+		color: 'red'
+	}
+});
+
+export default class ActionSheet extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			options: [],
+			height: 100
+		};
+		this.bottomSheetRef = React.createRef();
+	}
+
+	componentDidMount() {
+		EventEmitter.addEventListener(LISTENER, this.handleDisplay);
+	}
+
+	componentWillUnmount() {
+		EventEmitter.removeListener(LISTENER);
+	}
+
+	handleDisplay = ({ options, snapPoint }) => {
+		const height = options.length * buttonPaddingSize + options.length * textSize + bottomSheetHeaderHeight;
+		this.setState({ options, height });
+		this.bottomSheetRef.current.snapTo(snapPoint);
+		vibrate();
+	}
+
+	hideActionSheet = () => {
+		this.bottomSheetRef.current.snapTo(0);
+	}
+
+
+	renderHeader = () => (
+		<View style={styles.header}>
+			<View style={styles.panelHandle} />
+		</View>
+	);
+
+	buttonIcon = (icon, isDanger) => {
+		if (!icon) {
+			return null;
+		}
+		let dangerStyle = {};
+		if (isDanger) {
+			dangerStyle = styles.danger;
+		}
+		return (
+			<CustomIcon name={icon} size={18} style={[styles.panelButtonIcon, dangerStyle]} />
+		);
+	}
+
+	buttonText = (label, isDanger) => {
+		let dangerStyle = {};
+		if (isDanger) {
+			dangerStyle = styles.danger;
+		}
+		return (
+			<Text style={[styles.buttonText, dangerStyle]}>{label}</Text>
+		);
+	}
+
+	onOptionPress = (action) => {
+		action();
+		this.hideActionSheet();
+	}
+
+	renderInner = () => {
+		const { options, height } = this.state;
+		return (
+			<View style={[styles.panel, { height }]}>
+				{options.map(option => (
+					isIOS
```

Comment:
Is this necessary?
You should have been using `RectButton`

Reply:
yes https://github.com/osdnk/react-native-reanimated-bottom-sheet/issues/16